test/remote/gateways/remote_datatrans_test.rb

```diff
@@ -0,0 +1,187 @@
+require 'test_helper'
+
+class RemoteDatatransTest < Test::Unit::TestCase
+  def setup
+    @gateway = DatatransGateway.new(fixtures(:datatrans))
+
+    @amount = 756
+    @credit_card = credit_card('4242424242424242', verification_value: '123', first_name: 'John', last_name: 'Smith', month: 0o6, year: 2025)
+    @bad_amount = 100000 # anything grather than 500 EUR
+
+    @options = {
+      order_id: SecureRandom.random_number(1000000000).to_s,
+      description: 'An authorize',
+      email: 'john.smith@test.com'
+    }
+
+    @billing_address = address
+
+    @execute_threed = {
+      execute_threed: true,
+      redirect_url: 'http://www.example.com/redirect',
+      callback_url: 'http://www.example.com/callback',
+      three_ds_2: {
+        browser_info:  {
+          width: 390,
+          height: 400,
+          depth: 24,
+          timezone: 300,
+          user_agent: 'Spreedly Agent',
+          java: false,
+          javascript: true,
+          language: 'en-US',
+          browser_size: '05',
+          accept_header: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
+        }
+      }
+    }
+  end
+
+  def test_successful_authorize
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
```

Comment:
❓ Question:

Is there anything on the response that you could assert to confirm this was a purchase transaction, maybe something related to the settlement?

Reply:
The response is the same for auth and purchases,